File: AndroidDemo/studyproject/app/src/main/java/com/example/rxjavademo/RetrofitClient.java
package com.example.rxjavademo;

import retrofit2.Retrofit;
import retrofit2.adapter.rxjava3.RxJava3CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitClient {
    final String url = "https://jsonplaceholder.typicode.com/";
    private static RetrofitClient instance;
    private final AppService appService;

    // 私有构造函数
    private RetrofitClient() {
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(url)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava3CallAdapterFactory.create())
                .build();
        appService = retrofit.create(AppService.class);
    }
    // 获取单例实例
    public static synchronized RetrofitClient getInstance() {
        if (instance == null) {
            instance = new RetrofitClient();
        }
        return instance;
    }

    // 获取ApiService
    public AppService getApiService() {
        return appService;
    }
}

File: AndroidDemo/CoroutinesDemo/app/src/main/java/com/example/coroutinesdemo/RetrofitClient.kt
package com.example.coroutinesdemo

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object RetrofitClient {
    private const val BASE_URL = "https://www.baidu.com/"

    // 创建一个 OkHttpClient 实例，用于添加日志拦截器等配置
    private val client = OkHttpClient.Builder()
        .addInterceptor(HttpLoggingInterceptor().apply {
            level = HttpLoggingInterceptor.Level.BODY // 设置日志级别为BODY，以打印请求和响应信息
        })
        .build()

    // 创建 Retrofit 实例
    val retrofit: Retrofit = Retrofit.Builder()
        .baseUrl(BASE_URL)
        .client(client)
        .addConverterFactory(GsonConverterFactory.create()) // 使用 GsonConverter 处理 JSON 数据
        .build()
}